code/controllers/subsystem/processing/overlays.dm

```diff
@@ -0,0 +1,115 @@
+var/datum/subsystem/processing/overlays/SSoverlays
+
+/datum/subsystem/processing/overlays
+	name = "Overlays"
+	flags = SS_TICKER|SS_NO_INIT|SS_FIRE_IN_LOBBY|SS_KEEP_TIMING
+	wait = 1
+	priority = 500
+	can_fire = FALSE
+
+	stat_tag = "Ov"
+	currentrun = null
+
+/datum/subsystem/processing/overlays/New()
+	NEW_SS_GLOBAL(SSoverlays)
+
+/datum/subsystem/processing/overlays/Recover()
+	processing = SSoverlays.processing
+
+/datum/subsystem/processing/overlays/fire()
+	while(processing.len)
+		var/atom/thing = processing[processing.len]
+		processing.len--
+		if(thing)
+			if(LAZYLEN(thing.priority_overlays) && LAZYLEN(thing.our_overlays))
+				thing.overlays = (thing.our_overlays | thing.priority_overlays)
+			else if(LAZYLEN(thing.our_overlays))
+				thing.overlays = thing.our_overlays.Copy()
+			else if(LAZYLEN(thing.priority_overlays))			//Do these two have to be copied? We don't want assignments to them to trigger overlay updates. But, idk, bruh, BYOND. - Cyberboss
```

Comment:
I doubt you need to copy them, perhaps test this specifically because this will be a big cost to pay otherwise